engine/apps/mattermost/serializers.py

```diff
@@ -0,0 +1,48 @@
+from rest_framework import serializers
+from rest_framework.validators import UniqueTogetherValidator
+
+from apps.mattermost.client import MattermostClient
+from apps.mattermost.exceptions import MattermostAPIException, MattermostAPITokenInvalid
+from apps.mattermost.models import MattermostChannel
+from common.api_helpers.utils import CurrentOrganizationDefault
+
+
+class MattermostChannelSerializer(serializers.Serializer):
+    id = serializers.CharField(read_only=True, source="public_primary_key")
+    organization = serializers.HiddenField(default=CurrentOrganizationDefault())
+    channel_id = serializers.CharField()
+    channel_name = serializers.CharField()
+
+    class Meta:
+        model = MattermostChannel
+        fields = [
+            "id",
+            "organization",
+            "channel_id",
+            "channel_name",
+        ]
+        validators = [
+            UniqueTogetherValidator(queryset=MattermostChannel.objects.all(), fields=["organization", "channel_id"])
+        ]
+
+    def create(self, validated_data):
+        return MattermostChannel.objects.create(**validated_data)
+
+    def to_internal_value(self, data):
```

Comment:
Yeah my thought process was,

1. With bot token getting list of channels we need to store all of those channels and display the same in UI plus we have to manage the sync of those channels similar to slack implementation but that seems like a overkill at this point. We can have each channel to be integrated manually by the admin as this is just a one time activity and not expected to happen or change often
2. Yeah slash commands is also possible but again the admin has to provide the token for slash command to us using which we can verify the incoming slash command. 

I feel (2) to be a better option than (1). I don't feel much of a feasibility for the admin between (2) and current implementation since both the flows requires relatively similar manual effort. Let me know if you think otherwise implementing with slash commands is not that complicated [Ref](https://developers.mattermost.com/integrate/slash-commands/custom/).

Reply:
Right. If having a slash command does not imply a lot of changes, it sounds better, since that will be more consistent with the current telegram and msteams experience.